### ISSUE TITLE
Partial application of generic class gives _T@partial

### ISSUE THREAD

**Issue by NeilGirdhar** [microsoft/pyright#5157]
```python
from functools import partial
from typing import Any, Callable, Generic, TypeVar

T = TypeVar('T')

class C(Generic[T]):
    def __init__(self, fun: Callable[..., Any], *, x: int = 0):
        super().__init__()

    def g(self) -> None:
        pass


@partial(C, x=1)
def f() -> None:
    pass


reveal_type(f)  # "_T@partial"
f.g()  # error: Cannot access member "g" for type "object*"
```

**Comment by erictraut** [comment]
This is already fixed in the main branch and will be addressed in 1.1.310.

**Comment by NeilGirdhar** [comment]
Awesome, thank you!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.310, which I just published. It will also be included in a future release of pylance.
